packages/devtools-local-toolbox/src/clients/chrome.js

```diff
@@ -38,51 +40,70 @@ function connectClient() {
   const deferred = defer();
 
   if(!getValue("chrome.debug")) {
-    return Promise.resolve(createTabs([]))
+    return Promise.resolve(createTabs([], {}))
   }
 
-  const webSocketPort = getValue("chrome.webSocketPort");
-  const url = `http://localhost:${webSocketPort}/json/list`;
-  networkRequest(url).then(res => {
-    deferred.resolve(createTabs(JSON.parse(res.content)))
-  }).catch(() => deferred.reject());
+  const port = getValue("chrome.port");
+  function fetch(http, options, callback) {
+    const {host, path, port} = options;
+    const url = `http://${host}:${port}${path}`;
+    return networkRequest(url).then(res => callback(null, res.content))
+  }
```

Comment:
Yeah, it looks like a dirty hack to me... :)

Fact is, CRI is not meant to run in the browser, and actually I don't understand why you are doing so. I mean, if I were to write a web-based JavaScript debugger (DevTools like) I'd use CRI on the backend. I'm probably missing something and I've only briefly checked out debugger.html, but why do you need to run CRI in a web context?